### PR TITLE
feat: add pyroscope ingestion endpoint

### PR DESCRIPTION
**Pull request by korniltsev**
This is a poc introducing pyroscope's /ingest api for push mode.
It reuses pyrscope OSS's ingestionHandler and parsers, this allows to ingest all supported formats, like jfr, etc.
It is not supper efficient at the moment, for example for a pprof profile there are a number of conversions happening: 'serialized pprof' - 'pprof' - 'pyroscope trees' - 'pprof' - 'serialized pprof' and only then passed to the phlare ingestion service.

### CODE REVIEW

File: pkg/ingester/pyroscope/ingest_adapter.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			continue
+		}
+		series.Labels = append(series.Labels, &typesv1.LabelPair{
+			Name:  k,
+			Value: v,
+		})
+	}
+	series.Samples = []*pushv1.RawSample{{
+		RawProfile: b,
+		ID:         uuid.New().String(), //todo generate on a client
```
Comment (cyriltovena) [inline pkg/ingester/pyroscope/ingest_adapter.go:81]: it's fine to have it here. IMO

File: pkg/ingester/pyroscope/ingest_adapter.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	})
+	newString := func(value string) int64 {
+		id := int64(len(pprof.StringTable))
+		pprof.StringTable = append(pprof.StringTable, value)
+		return id
+	}
+	// todo(korniltsev) the pprof is partially built by pyroscope oss and partially here, do it in one place
+	pprof.Mapping = []*tree.Mapping{{Id: 0}} // a fake mapping
+	if pi.SampleRate != 0 && (metric == metricWall || metric == metricProcessCPU) {
+		pprof.Period = 1_000_000_000 / int64(pi.SampleRate)
```
Comment (petethepig) [inline pkg/ingester/pyroscope/ingest_adapter.go:75]: it's not obvious what `1_000_000_000` is

**Comment by petethepig** [comment]
Talked to @korniltsev — he's going to be finishing this PR today.

File: pkg/phlare/modules.go [hunk 3]
```diff
@@ -364,12 +365,15 @@ func (f *Phlare) context() context.Context {
...
 
-	ingester, err := ingester.New(f.context(), f.Cfg.Ingester, f.Cfg.PhlareDB, f.storageBucket, f.Overrides)
+	svc, err := ingester.New(f.context(), f.Cfg.Ingester, f.Cfg.PhlareDB, f.storageBucket, f.Overrides)
 	if err != nil {
 		return nil, err
 	}
-	ingesterv1connect.RegisterIngesterServiceHandler(f.Server.HTTP, ingester, f.auth)
-	return ingester, nil
+	ingesterv1connect.RegisterIngesterServiceHandler(f.Server.HTTP, svc, f.auth)
+	//pyroscopePath := "/pyroscope/ingest"
```
Comment (petethepig) [inline pkg/phlare/modules.go:373]: @cyriltovena @simonswine Not sure how you all feel about the prefix? I know /render and /label-values [have the prefix](https://github.com/grafana/phlare/pull/591/files#diff-ff08860a084147a57eed45febf9519697bc6179c98981010040b66ffde458e79R208-R209)

I don't feel strongly one way or the other.

Comment (cyriltovena) [reply pkg/phlare/modules.go:373]: Fine for me

File: pkg/phlaredb/delta.go [hunk 4]
```diff
@@ -97,12 +100,21 @@ func copySample(s *schemav1.Sample) *schemav1.Sample {
 }
 
 func isDelta(lbs phlaremodel.Labels) bool {
+	if lbs.Get(phlaremodel.LabelNameDelta) == "false" {
```
Comment (cyriltovena) [inline pkg/phlaredb/delta.go:103]: Nit

It's a bit confusing that you can only use the label to deactivate.

May be if the labels is true then we should return true too and if not a Boolean like empty go to the rest of the function.

File: pkg/phlare/modules.go (hunk 3 above)
Comment (petethepig) [reply pkg/phlare/modules.go:373]: FWIW, I added `/pyroscope` prefix [here](https://github.com/grafana/backend-enterprise/pull/4418/files#diff-ef48fc318181e7feb0e84c93de69f745eec72eb9450764e97a15866530656c88R370) for consistency, so I guess now I do lean slightly more towards having the prefix.

**Review by cyriltovena — APPROVED** [review]
Lgtm

File: pkg/phlare/modules.go [hunk 5]
```diff
@@ -364,12 +365,15 @@ func (f *Phlare) context() context.Context {
...
+	pyroscopePath := "/ingest"
```
Comment (petethepig) [inline pkg/phlare/modules.go:374]:
```suggestion
	pyroscopePath := "/pyroscope/ingest"
```

**Pull request merged**